end2end/tests/reportBuilder.spec.ts

```diff
@@ -2,30 +2,208 @@ import { test, expect } from '@playwright/test';
+    await page.getByLabel('comment text area').fill('testing purpose');
+    await page.getByRole('button', { name: 'Close' }).click();
+
+    // un-comment if need to Approve Action 
+    // await page.getByRole('button', { name: 'Approve' }).click();
+});
+
+test("Test Share Report button", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('button', { name: 'Share Report' }).click();
+    await page.getByText('https://host.docker.internal/').click();
+    await page.getByText('This link can be shared to provide a live view into this report.https://host.').click();
+    await page.getByRole('button', { name: 'Close' }).click();
+    //await page.getByRole('button', { name: 'Email Report', exact: true }).click();
+});
+
+test("Test Edit Labels/JSON button working", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHccBjAkYDUQYTQYNCjEAEZbdPJ4xLAAeQAzPLh9OxUgA&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgWJ4RVFABCk5Giiz6jRdWWqeAXSA%3D");
+    await page.getByRole('button', { name: 'JSON' }).click();
+    await page.getByRole('button', { name: 'Shorten Link' }).click();
+    await page.getByRole('button', { name: 'Expand Link' }).click();
+    await page.getByRole('button', { name: 'Close' }).click();
+});
+
+test('validate create row button', async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/")
+    await page.getByText('Report Builder Create custom').click();
+    await page.getByRole('cell', { name: 'Current Status' }).locator('a').click();
+    await page.getByRole('option', { name: 'Type' }).click();
+    await page.getByRole('cell', { name: 'Complex Form' }).locator('a').click();
+    await page.getByRole('option', { name: 'General Form' }).click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.getByText('Type of Request').click();
+    await page.getByText('Action Button').click();
+    await page.locator('#indicatorList').getByText('General Form').click();
+    await page.getByText('Assigned Person 2').click();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+    await expect(page.getByRole('button', { name: 'Create Row' })).toBeVisible();
+});
+
+test('modify search with And logical filter', async ({ page }) => {
```

Comment:
The label indicates that it's testing a logical filter. The assertion should check if the logical filter works. We should expect to see a record within the result that matches the conditions.

Reply:
Resolved